accounts-db/benches/bench_lock_accounts.rs

```diff
@@ -0,0 +1,98 @@
+#![feature(test)]
+#![allow(clippy::arithmetic_side_effects)]
+
+extern crate test;
+
+use {
+    criterion::{criterion_group, criterion_main, Criterion},
+    rand::{seq::SliceRandom, thread_rng},
+    solana_accounts_db::{accounts::Accounts, accounts_db::AccountsDb},
+    solana_sdk::{
+        account::AccountSharedData,
+        instruction::{AccountMeta, Instruction},
+        pubkey::Pubkey,
+        system_program,
+        transaction::{SanitizedTransaction, Transaction, MAX_TX_ACCOUNT_LOCKS},
+    },
+    std::sync::Arc,
+};
+
+// simultaneous transactions locked
+const BATCH_SIZES: [usize; 3] = [1, 32, 64];
+
+// locks acquired per transaction
+const LOCK_COUNTS: [usize; 2] = [2, 64];
+
+// largest batch size * largest lock count * 2
+const ACCOUNTS_DB_SIZE: usize = 8192;
+
+fn create_test_data(batch_size: usize, lock_count: usize) -> (Accounts, Vec<SanitizedTransaction>) {
+    let accounts_db = AccountsDb::new_single_for_tests();
+    let accounts = Accounts::new(Arc::new(accounts_db));
+
+    let mut all_account_pubkeys = vec![];
+    let mut transactions = vec![];
+
+    for _ in 0..batch_size {
+        let mut account_metas = vec![];
+
+        // `lock_accounts()` distinguishes writable from readonly, so give transactions an even split
+        // signer doesnt matter for locking but `sanitize()` expects to see at least one
```

Comment:
Even split seems fine here to me.
I wonder if we should add a 3rd param to the parameter suite for whether or not these read accounts conflict.
If so, just do all conflicting / all not conflicting.

Doubles the number of benchmarks, but think these should be relatively quick with only a few